### PR TITLE
User follow and unfollow feature

### PR DESCRIPTION
**Pull request by burakcankazim**
Added necessary endpoints to implement the function of following and unfollowing other users.

### CODE REVIEW

**Comment by burakcankazim** [comment]
Merged and solved conflict of other two pr's that are waiting to be merged, can be merged last safely.

File: application/backend/src/main/java/cmpe451/group12/beabee/goalspace/controller/UserController.java [hunk 1]
```diff
@@ -40,4 +41,32 @@ public UserGetDTO getUser(@PathVariable @ApiParam(value = "Search query.", examp
...
+
+    @ApiOperation(value = "Get followers of a user.")
+    @GetMapping("/{userId}/followers")
+    public List<UserGetDTO> getFollowers(@PathVariable @ApiParam(value = "Id of the user.", example = "5") Long userId) {
+        return userService.getFollowers(userId);
+    }
+
+
+    @ApiOperation(value = "Get followers of a user.")
+    @GetMapping("/{userId}/followings")
```
Comment (veyisTurgut) [inline application/backend/src/main/java/cmpe451/group12/beabee/goalspace/controller/UserController.java:68]: todo:  typo

**Review by veyisTurgut — CHANGES_REQUESTED** [review]

File: application/backend/src/main/java/cmpe451/group12/beabee/goalspace/controller/UserController.java (hunk 1 above)
Comment (veyisTurgut) [reply application/backend/src/main/java/cmpe451/group12/beabee/goalspace/controller/UserController.java:68]: I asked the change for a different file. But it shows this typo, let me check again.

File: application/backend/src/main/java/cmpe451/group12/beabee/common/mapper/UserMapper.java [hunk 2]
```diff
@@ -2,19 +2,44 @@
...
+                .userCredentials(UserCredentialsGetDTO.builder()
+                        .user_id(user.getUser_id())
+                        .username(user.getUsername())
+                        .email(user.getEmail())
+                        .password("****")
+                        .build())
+                .build();
+    }
+
+    // TODO Do we really need this for GetDTO
```
Comment (veyisTurgut) [inline application/backend/src/main/java/cmpe451/group12/beabee/common/mapper/UserMapper.java:28]: You are right, it's never used. But it hurts nobody. Let it stay.

File: application/backend/src/main/java/cmpe451/group12/beabee/common/service/UserService.java [hunk 3]
```diff
@@ -118,4 +120,66 @@ public UserAnalyticsDTO getAnalytics(Long user_id) {
...
+            return new MessageResponse("Not following!", MessageType.ERROR);
+        }
+
+        user.getFollowing().remove(targetUser);
+        targetUser.getFollowers().remove(user);
+
+        userRepository.save(user);
+        userRepository.save(targetUser);
+
+        activityStreamService.unfollowUserSchema(user, targetUser);
```
Comment (veyisTurgut) [inline application/backend/src/main/java/cmpe451/group12/beabee/common/service/UserService.java:163]: There is no UNFOLLOW activity type in activity stream vocab: https://www.w3.org/TR/activitystreams-vocabulary/#activity-types . Either remove the unfollowUserSchema or change activityType to "REMOVE".

**Review by veyisTurgut — CHANGES_REQUESTED** [review]

File: application/backend/src/main/java/cmpe451/group12/beabee/common/service/UserService.java (hunk 3 above)
Comment (burakcankazim) [reply application/backend/src/main/java/cmpe451/group12/beabee/common/service/UserService.java:163]: Removing it for now

**Pull request merged**